packages/commonwealth/client/scripts/views/components/reaction_button/ReactionButtonLogic.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import TopicGateCheck from '../../../controllers/chain/ethereum/gatedTopic';
+import type { ChainInfo, Thread } from '../../../models/index';
+import app from '../../../state';
+
+export const ReactionButtonLogic = (thread: Thread, setReactors) => {
+  const activeAddress = app.user.activeAccount?.address;
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const thisUserReaction = thread.associatedReactions.filter(r => r.address === activeAddress);
+  const [hasReacted, setHasReacted] = useState(thisUserReaction.length !== 0);
+  const [reactedId, setReactedId] = useState(thisUserReaction.length === 0 ? -1 : thisUserReaction[0].id);
+
+  useEffect(() => {
+    const fetch = () => {
+      if (app.user.activeAccount && thread.associatedReactions.filter((r) => r.address === activeAddress).length > 0) {
+        setHasReacted(true);
+      } else {
+        setHasReacted(false);
+      }
+
+      setReactors(thread.associatedReactions.map(t => t.address));
+    };
+
+    fetch();
+  }, []);
```

Comment:
missing dependencies 
![image](https://user-images.githubusercontent.com/14819225/227277470-ede27f25-c4df-4d1d-b74c-f40247f5affa.png)

Reply:
This needs to be eslint-ignored because the way I am handling threads.associatedReactions. If this is run before the response of the post request to the server, it will revert to not displaying the newly created reaction.